### DISCUSSION TITLE
Warning: No tags are present for this repository

### DISCUSSION THREAD

**Discussion by lindseyschuler** [PaulHatch/semantic-version#107] [Q&A]
Hi,

I'm trying to implement this action in our repo and no matter what I try I get the following warning:

No tags are present for this repository

I'm running a checkout first and see the latest tag in the output and if I run a $ git tag --points-at HEAD after I do a checkout I see the latest tag below
v5.1.0

Below is my action
 - name: get semantic version
      id: sem-ver
      uses: paulhatch/semantic-version@v5.0.2
      with:
        tag_prefix: v
        use_branches: true
        major_pattern: (MAJOR)
        minor_pattern: (MINOR)
        bump_each_commit: true

No matter what I do I get the same error every time. Am I missing something in my setup?

**Comment by PaulHatch** [comment]
So if you are using `bump_each_commit: true` you won't necessarily have tags, as the version is bumped without them. You are also using branches to version, this is an unusual setup TBH, I don't think I've ever combined these two. Generally branches would only be used if you were supporting ongoing releases to multiple versions while bump each commit is generally used for small projects that have a less formal release process.

There is an open ticket for enhancing support for branches, since branches commits change and since multiple branches can be present at the same time there are a number of edge cases that require opinionated decisions to be made. At the moment I'd recommend sticking with tags unless the current branching behavior is preferable or necessary.

Having said all that, if you have versions that you expect to be used to determine the version you can check that:
- Make sure the git history is being pulled [as described here](https://github.com/PaulHatch/semantic-version#important-note-regarding-the-checkout-action). By default the `actions/checkout` only pulls the most recent commit.
- Make sure your tag or branch format matches the expected format. By default this is like `v1.0.0`.

If you're still having trouble, follow the steps described [here](https://github.com/PaulHatch/semantic-version/issues/117#issuecomment-1694349525) and post the resulting diagnostic information.
